catalog/service/rest/src/main/java/org/projectnessie/catalog/service/rest/IcebergApiV1GenericResource.java

```diff
@@ -138,14 +145,33 @@ public Uni<Void> commitTransaction(
     commitTransactionRequest.tableChanges().stream()
         .map(
             tableChange -> {
+              List<IcebergMetadataUpdate> updates =
+                  tableChange.updates().stream()
+                      .filter(update -> !(update instanceof SetLocation))
+                      .collect(Collectors.toList());
+
               ContentKey key = requireNonNull(tableChange.identifier()).toNessieContentKey();
 
-              return IcebergCatalogOperation.builder()
-                  .updates(tableChange.updates())
-                  .requirements(tableChange.requirements())
-                  .key(key)
-                  .type(ICEBERG_TABLE)
-                  .build();
+              IcebergCatalogOperation.Builder builder =
+                  IcebergCatalogOperation.builder()
+                      .updates(updates)
+                      .requirements(tableChange.requirements())
+                      .key(key)
+                      .warehouse(decoded.warehouse())
+                      .type(ICEBERG_TABLE);
+
+              if (tableChange.hasAssertCreate()) {
+                List<IcebergUpdateRequirement> invalidRequirements =
+                    tableChange.requirements().stream()
+                        .filter(req -> !(req instanceof AssertCreate))
+                        .collect(Collectors.toList());
+                checkArgument(
```

Comment:
Don't we already do this implicitly via the `tableExists`/`viewExists` parameter in `IcebergUpdateRequirement`?